src/input_cp2k_global.F

```diff
@@ -99,28 +107,57 @@ SUBROUTINE create_global_section(section)
 
       CALL keyword_create(keyword, name="BLACS_REPEATABLE", &
                           description="Use a topology for BLACS collectives that is guaranteed to be repeatable "// &
-                          "on homegeneous architectures", &
+                          "on homogeneous architectures", &
                           usage="BLACS_REPEATABLE", &
                           default_l_val=.FALSE., lone_keyword_l_val=.TRUE.)
       CALL section_add_keyword(section, keyword)
       CALL keyword_release(keyword)
 
+      CALL GET_ENVIRONMENT_VARIABLE("CP2K_DIAG_LIBRARY", env, STATUS=env_stat)
+      IF (0 .NE. env_stat) THEN
+#if defined(__ELPA)
+         default_i_val = do_diag_elpa
+#elif defined(__SCALAPACK2)
+         default_i_val = do_diag_sl2
+#else
+         default_i_val = do_diag_sl
+#endif
+      ELSE
+         READ (env, "(I32)") default_i_val
+      END IF
+
       CALL keyword_create( &
          keyword, name="PREFERRED_DIAG_LIBRARY", &
-         description="Specifies the DIAGONALIZATION library to be used. If not availabe, the standard scalapack is used", &
+         description="Specifies the DIAGONALIZATION library to be used. If not available, the standard scalapack is used", &
          usage="PREFERRED_DIAG_LIBRARY ELPA", &
-         default_i_val=do_diag_sl, &
+         default_i_val=default_i_val, &
          enum_i_vals=(/do_diag_sl, do_diag_sl2, do_diag_elpa/), &
          enum_c_vals=s2a("SL", "SL2", "ELPA"), &
          enum_desc=s2a("Standard scalapack: syevd", "Scalapack 2.0: syevr", "ELPA"), &
          citations=(/Marek2014/))
       CALL section_add_keyword(section, keyword)
       CALL keyword_release(keyword)
 
+      default_i_val = 1 ! AUTO (ELPA-kernel)
+#if defined(__ELPA)
+      IF (m_cpuid() .LE. MACHINE_X86) THEN
+         SELECT CASE (m_cpuid ())
+         CASE (MACHINE_X86_SSE4)
+            default_i_val = 8 ! ELPA_2STAGE_REAL_SSE_BLOCK4
```

Comment:
Currently, the enumeration of the ELPA kernels is fully encapsulated in `cp_fm_elpa.F`. Maybe we should keep it that way and add a routine `get_elpa_kernel_index_for_name()` ?

Reply:
Good idea!  I also need to look at PR #329....

I will fix the name of the environment variable to match the input spec./keyword. If it does not outgrow, I will also check if the keyword parser can be reused for the environment variable. Later, I think it's worth to check if every global keyword can be optionally accompanied by an environment variable of the same name.

Reply:
@hfp should I go ahead with merging PR #329 or do I wait for yours?

Reply:
Please go ahead!